engine/src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -285,16 +290,40 @@ object CromwellApiService {
   import spray.json._
 
   implicit class EnhancedThrowable(val e: Throwable) extends AnyVal {
-    def failRequest(statusCode: StatusCode): Route = complete((statusCode, APIResponse.fail(e).toJson.prettyPrint))
-    def errorRequest(statusCode: StatusCode): Route = complete((statusCode, APIResponse.error(e).toJson.prettyPrint))
+    def failRequest(statusCode: StatusCode, warnings: Seq[String] = Vector.empty): Route = {
+      completeResponse(statusCode, APIResponse.fail(e).toJson.prettyPrint, warnings)
+    }
+    def errorRequest(statusCode: StatusCode, warnings: Seq[String] = Vector.empty): Route = {
+      completeResponse(statusCode, APIResponse.error(e).toJson.prettyPrint, warnings)
+    }
+  }
+
+  def completeResponse[A](statusCode: StatusCode, value: A, warnings: Seq[String])
+                         (implicit mt: ToEntityMarshaller[A]): Route = {
+    val warningHeaders = warnings.toIndexedSeq map { warning =>
+      /*
+      Need a quoted string.
+      https://stackoverflow.com/questions/7886782
+
+      Using a poor version of ~~#!
+      https://github.com/akka/akka-http/blob/v10.0.9/akka-http-core/src/main/scala/akka/http/impl/util/Rendering.scala#L206
+       */
+      val quotedString = "\"" + warning.replaceAll("\"","\\\\\"").replaceAll("[\\r\\n]+", " ").trim + "\""
+
+      // https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.46
+      RawHeader("Warning", s"299 cromwell/$cromwellVersion $quotedString")
```

Comment:
ToL: Is this actually looked at? 

This feels a little bit like `... > /dev/null`